tests/unit/test-firefox/test.firefox.js

```diff
@@ -324,6 +324,149 @@ describe('firefox', () => {
       }
     ));
 
+    it('configures a named profile', () => withTempDir(
+      (tmpDir) => {
+        const configureProfile =
+          sinon.spy((profile) => Promise.resolve(profile));
+        const app = 'fennec';
+        const profilePath = tmpDir.path();
+        const profileFinder = () => {
```

Comment:
@harikishen `profileFinder` is a function that creates the fake `Finder` objects, and not a `Finder` object itself, that is the reason why the assertion commented on line 351 is not working for you.

You should do something like:
```js
const profileFinder = {
  getPath: sinon.spy(...),
  ...
};

const createProfileFinder = () => profileFinder;

const profile = await firefox.useProfile(profilePath, {
   ...
   configureThisProfile: configureProfile,
   createProfileFinder: createProfileFinder,
});

assert.equal(configureProfile.called, true);
...
assert.equal(profileFinder.getPath.callCount, 2)
```